tools/manifest.json

```diff
@@ -73,5 +84,30 @@
 			},
 			"ValidFor": 7
 		}
+	},
+	"TLS": {
+		"web": {
+			"InternalClient": [
```

Comment:
I'm not quite happy with these names. Suggestions?

Reply:
For each label, you want to define incoming and outgoing connections that should be wrapped in TLS, correct?
Do you further want to support mesh-external mTLS? That's fine with me, but be aware this is a feature none of the regular service meshes support.

Based on my understanding I'd suggest something like this:
```
"mTLS":  {
    "label": {
        "Ingress": [
             {"port": 8080, "addr": "*", external: True},
             {"port": 8080, "label": "svc"},
        ],
        "Egress": [
             {"port": 8080, "addr": "label"},
             ...
         ]
        
    }
}
```
This is just a simple draft, but the key points are:
* Each marble can have several mTLS labels
* For each label define Ingress and Egress
* Ingress/Egress include port, fqdn or label, external/internal
* port, fqdn, and label can be wildcards

What is the default behavior? No mTLS? Block connection? OSM by default blocks all connections but has a permissive mode which can be enabled during the install.

Reply:
> For each label, you want to define incoming and outgoing connections that should be wrapped in TLS, correct?

yes

> Do you further want to support mesh-external mTLS?

No, just external TLS. E.g., provide TLS config for the emojivoto HTTP web server.

Can you explain what Ingress and Egress means in terms of marblerun running without k8s?
What is label instead of addr? (How) can this be implemented (securely)?

Reply:
> No, just external TLS. E.g., provide TLS config for the emojivoto HTTP web server.

Ok, so we would handle incoming TLS traffic unwrap and forward the payload. Makes sense, but in most cases, users want to define their own certificate here, so I would add the option to reference a secret, which is probably what `_comment` is meant for.

> Can you explain what Ingress and Egress means in terms of marblerun running without k8s?

Yes, `Ingress` and `Egress` are k8s terms, you could also use `Incoming` and `Outgoing` or smth, but they seem to be fitting terms for incoming and outgoing traffic. If used without k8s you'd still have the same two groups, but maybe the naming is confusing?

> What is label instead of addr? (How) can this be implemented (securely)?

Say you'd like to express that mTLS should be applied between the `web` and `voting-svc`.
You could do the following:

```
"mTLS":  {
    "web": {
        "Egress": [
             {"port": 8080, "addr": "voting-svc.emojivoto"},
         ]
    },
    "voting-svc": {
        "Ingress": [
             {"port": 8080, "addr": "web.emojivoto"},
         ]
    }
}
```
But this involves knowing the fqdn of at least the `voting-svc`. It somehow mixes the abstract mTLS definition with the underlying network configuration. To me it would seem cleaner if you could just use the mTLS labels instead:

```
"mTLS":  {
    "web": {
        "Egress": [
             {"port": 8080, "label": "voting-svc"},
         ]
    },
    "voting-svc": {
        "Ingress": [
             {"port": 8080, "label": "web"},
         ]
    }
}
```